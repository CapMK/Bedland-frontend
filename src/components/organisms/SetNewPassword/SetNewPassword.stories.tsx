import type { Meta, StoryObj } from '@storybook/react';
import { BrowserRouter } from 'react-router-dom';
import GlobalStyle from 'theme/globalStyles';
import ThemeProvider from 'theme/ThemeContext';
import SetNewPassword from './SetNewPassword';

// docs: https://storybook.js.org/docs/7.0/react/writing-stories/introduction
const meta = {
  title: 'Organisms/SetNewPassword',
  component: SetNewPassword,
  tags: ['autodocs'],
  decorators: [
    (Story) => (
      <ThemeProvider>
        <GlobalStyle />
        <BrowserRouter>{Story()}</BrowserRouter>
      </ThemeProvider>
    ),
  ],
} satisfies Meta<typeof SetNewPassword>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {};
