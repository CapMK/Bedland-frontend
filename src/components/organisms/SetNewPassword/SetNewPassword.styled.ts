import styled from 'styled-components';
import { PaletteType } from 'theme/theme.types';

export const SetNewPasswordContainer = styled.div<{ palette: PaletteType }>`
  color: ${({ palette }) => (palette as PaletteType).light};
  display: flex;
  flex-direction: column;
  gap: 1.5rem;
  height: 27.6875rem;
  justify-content: center;
  width: 25.625rem;
`;
export const ResPassCenteredHeading = styled.h1`
  display: flex;
  justify-content: center;
`;
