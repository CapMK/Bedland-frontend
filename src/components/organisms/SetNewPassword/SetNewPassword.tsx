import { useContext } from 'react';

import RulesDescription from 'components/molecules/PasswordRulesDescription/RulesDescription';
import ResetPasswordForm from 'components/organisms/ResetPasswordForm/ResetPasswordForm';
import { ThemeContext } from 'theme/ThemeContext';
import * as Styled from './SetNewPassword.styled';

const SetNewPassword = () => {
  const { palette } = useContext(ThemeContext);

  return (
    <Styled.SetNewPasswordContainer palette={palette}>
      <Styled.ResPassCenteredHeading>Reset your password</Styled.ResPassCenteredHeading>
      <RulesDescription />
      <ResetPasswordForm />
    </Styled.SetNewPasswordContainer>
  );
};
export default SetNewPassword;
