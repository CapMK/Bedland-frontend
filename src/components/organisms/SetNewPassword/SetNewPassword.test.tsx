import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import ThemeProvider from 'theme/ThemeContext';
import SetNewPassword from './SetNewPassword';

describe('SetNewPassword', () => {
  afterEach(cleanup);
  it('renders correctly', () => {
    const snapshot = renderer
      .create(
        <ThemeProvider>
          <BrowserRouter>
            <SetNewPassword />
          </BrowserRouter>
        </ThemeProvider>,
      )
      .toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
