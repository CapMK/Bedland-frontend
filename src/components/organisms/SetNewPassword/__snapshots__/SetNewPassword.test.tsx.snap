// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SetNewPassword > renders correctly 1`] = `
.c2 {
  padding: 0rem 2px 0rem 2.9375rem;
  width: 100%;
}

.c3 {
  liststyletype: disc;
  margin-left: 10px;
  padding: 0 0 0 1rem;
}

.c8 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 2.875rem;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  width: 100%;
}

.c7 {
  border: 1px solid #919AFF;
  border-radius: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 2.875rem;
  padding: 0.625rem 1rem 0.625rem 1rem;
  width: 19.875rem;
}

.c7:disabled,
.c7:hover,
.c7:active,
.c7:focus {
  outline: none;
}

.c6 {
  color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 16;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.6875rem;
  width: 19.875rem;
}

.c9 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  border: 1px solid transparent;
  border-radius: 0.5rem;
  color: #FF7C33;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 10px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5625rem 1rem;
  white-space: nowrap;
  width: 100%;
  font-family: Source Sans Pro,sans-serif;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1.4;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

.c9:hover {
  background-color: #FF7C330D;
  border: 1px solid transparent;
}

.c10 {
  background-color: transparent;
  border: 1.75px solid transparent;
  color: #FF7C33;
  padding: undefinedrem 1rem;
}

.c10:hover {
  background-color: #FF7C330D;
  border: 1.75px solid transparent;
}

.c10:focus {
  background-color: #FF7C3326;
  border: 1.75px solid transparent;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1rem;
  padding: 0 2.84375rem 0rem 2.84375rem;
  width: 100%;
}

.c0 {
  color: #FCFEFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1.5rem;
  height: 27.6875rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 25.625rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  className="c0"
>
  <h1
    className="c1"
  >
    Reset your password
  </h1>
  <div
    className="c2"
  >
    <span>
       Remember, your password must include:
    </span>
     
    <ol
      className="c3"
      style={
        {
          "listStyleType": "disc",
        }
      }
    >
      <li>
        8-24 characters
      </li>
      <li>
        At least one uppercase character
      </li>
      <li>
        At least one special character
      </li>
    </ol>
  </div>
  <form
    className="c4"
    onSubmit={[Function]}
  >
    <div
      className="c5"
    >
      <label
        className="c6"
        htmlFor="password"
        size="medium"
      >
        Your new password
      </label>
      <input
        className="c7"
        data-testid="password"
        id="password"
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Enter your new password"
        type="password"
      />
    </div>
    <div
      className="c5"
    >
      <label
        className="c6"
        htmlFor="confirmPassword"
        size="medium"
      >
        Confirm the password
      </label>
      <input
        className="c7"
        data-testid="confirmPassword"
        id="confirmPassword"
        name="confirmPassword"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Confirm the password"
        type="password"
      />
    </div>
    <span
      className="c8"
    />
    <button
      className="c9 c10"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      type="submit"
    >
      Reset my password
    </button>
  </form>
</div>
`;
